rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,526 @@
+        expected_steth_transfer_index = context.tx_log.log_index - 1
+        expected_wsteth_transfer_index = context.tx_log.log_index - 2
+        sth_received_amount = None
+        wsteth_unwrapped_amount = None
+        for tx_log in context.all_logs:
+            if (
+                tx_log.log_index == expected_steth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == steth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == wsteth_evm_address and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == user_address  # to # wsteth
+            ):
+                sth_received_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_STETH.resolve_to_crypto_asset(),
+                )
+            elif (
+                tx_log.log_index == expected_wsteth_transfer_index and
+                tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                tx_log.address == wsteth_evm_address and
+                hex_or_bytes_to_address(tx_log.topics[1]) == user_address and  # from
+                hex_or_bytes_to_address(tx_log.topics[2]) == ZERO_ADDRESS  # to # wsteth
+            ):
+                wsteth_unwrapped_amount = asset_normalized_value(
+                    amount=hex_or_bytes_to_int(tx_log.data[:32]),
+                    asset=A_WSTETH.resolve_to_crypto_asset(),
+                )
+
+            if sth_received_amount and wsteth_unwrapped_amount:
+                paired_wsteth_ratio = FVal(sth_received_amount) / FVal(wsteth_unwrapped_amount)
+                break
+
+        if paired_wsteth_ratio is None:
+            return DEFAULT_DECODING_OUTPUT
+        in_event = None
+        out_event = None
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_WSTETH and
+                event.balance.amount == wsteth_unwrapped_amount and
+                event.address == ZERO_ADDRESS
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.counterparty = CPT_LIDO_ETH
+                event.notes = f'Sent {wsteth_unwrapped_amount} {A_WSTETH.symbol_or_name()} for unwrapping at wstETH ratio {paired_wsteth_ratio}'  # noqa: E501
+                out_event = event
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_STETH and
+                event.balance.amount == sth_received_amount and
+                event.address == wsteth_evm_address
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL  # TODO: decide the right mappings
```

Comment:
Rotki team: I would like to we discuss with you about how to map the lido events, because it is a staking, but users receive tokens in return that have a deep market, and often they are exchanged without returning the tokens to the Lido contract, so I am not sure how to map (I have never used the STAKING event type for this reason)

Reply:
I would map it with same events as we do in compound, aave and other tokens. Not as staking. So same events type/subtypes as we use in compound, aave etc for wrapping and unwrapping a token